crates/matrix-sdk-ui/tests/integration/timeline/pagination.rs

```diff
@@ -842,3 +849,246 @@ async fn test_back_pagination_aborted() {
+    let event_factory = EventFactory::new().room(room_id).sender(&ALICE);
+
+    let mock_server = MatrixMockServer::new().await;
+    let client = mock_server.client_builder().build().await;
+
+    let room = mock_server.sync_joined_room(&client, room_id).await;
+    let timeline = room.timeline().await.unwrap();
+    let (timeline_initial_items, mut timeline_stream) = timeline.subscribe().await;
+
+    assert!(timeline_initial_items.is_empty());
+    assert_pending!(timeline_stream);
+
+    // Receive 30 events.
+    mock_server
+        .mock_sync()
+        .ok_and_run(&client, |sync_builder| {
+            sync_builder.add_joined_room({
+                let mut room = JoinedRoomBuilder::new(room_id);
+
+                for nth in 0..30 {
+                    room = room.add_timeline_event(
+                        event_factory
+                            .text_msg("foo")
+                            .event_id(&EventId::parse(format!("$ev{nth}")).unwrap()),
+                    );
+                }
+
+                room.set_timeline_prev_batch("back_pagination_token_1").set_timeline_limited()
+            });
+        })
+        .await;
+
+    // Only 20 items are broadcasted to the subscriber.
+    assert_timeline_stream! {
+        [timeline_stream]
+        // Huh?
+        // TODO: understand why
+        clear;
+        clear;
+
+        // `$ev11`, the 1st item, and its update.
+        append "$ev11";
+        update[0] "$ev11";
+
+        // `$ev12` and its update.
+        append "$ev12";
+        update[1] "$ev12";
+
+        // and so on…
+        append "$ev13";
+        update[2] "$ev13";
+
+        append "$ev14";
+        update[3] "$ev14";
+
+        append "$ev15";
+        update[4] "$ev15";
+
+        append "$ev16";
+        update[5] "$ev16";
+
+        append "$ev17";
+        update[6] "$ev17";
+
+        append "$ev18";
+        update[7] "$ev18";
+
+        append "$ev19";
+        update[8] "$ev19";
+
+        append "$ev20";
+        update[9] "$ev20";
+
+        append "$ev21";
+        update[10] "$ev21";
+
+        append "$ev22";
+        update[11] "$ev22";
+
+        append "$ev23";
+        update[12] "$ev23";
+
+        append "$ev24";
+        update[13] "$ev24";
+
+        append "$ev25";
+        update[14] "$ev25";
+
+        append "$ev26";
+        update[15] "$ev26";
+
+        append "$ev27";
+        update[16] "$ev27";
+
+        append "$ev28";
+        update[17] "$ev28";
+
+        // … until we receive `$ev29`: the last received event, and the 19th item.
+        append "$ev29";
+
+        // The day divider is inserted before `$ev0`, so it shifts items to
+        // the right: `$ev10` becomes part of the stream.
+        //
+        // This is the 20th item, hurray!
+        prepend "$ev10";
+    };
+
+    // Nothing more!
+    assert_pending!(timeline_stream);
+
+    let no_network_pagination =
+        mock_server.mock_room_messages().error500().never().mount_as_scoped().await;
```

Comment:
nit: you do not need to define it: if the SDK tried to reach it, it would then run into a 404, equivalently. (And then you wouldn't need the `drop(no_network_pagination)` below either!)

Reply:
Oh yeah, sure…